rust/src/ssh/ssh.rs

```diff
@@ -179,8 +214,16 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
-                                hdr.flags = SSHConnectionState::SshStateFinished;
+                            match head.msg_code { 
+                                parser::MessageCode::SshMsgNewKeys => {
+                                    hdr.flags = SSHConnectionState::SshStateFinished;
+                                }
+                                parser::MessageCode::SshMsgKexinit if HASSH_ENABLED.load(Ordering::Relaxed) => {
+                                    // saving kex message for future hassh generation
+                                    hdr.record_left_buf.extend_from_slice(&rem);
```

Comment:
Thanks for handling messages other multiple packets.

But there is no need to have the buffer in the Rust code, as there is a buffer in the TCP upper layer.
To use it, the rust function should return `AppLayerResult::incomplete` saying how much more data it needs.
This should also have a limit on the buffer size, so as not to extend the buffer up to 4Gbytes of memory (SMB protocol does that)

Reply:
Is it ok to put a limit of 65535 bytes as a maximum size?